pkg/policy/selectorcache.go

```diff
@@ -55,6 +57,24 @@ type CachedSelector interface {
 // CachedSelectorSlice is a slice of CachedSelectors that can be sorted.
 type CachedSelectorSlice []CachedSelector
 
+// MarshalJSON returns the CachedSelectors as JSON formatted buffer
+func (s CachedSelectorSlice) MarshalJSON() ([]byte, error) {
```

Comment:
Out of curiosity, why do we need a custom Marshaler?

Reply:
It was because currently, it's possible that an implementation of the `CachedSelector` interface doesn't export the relevant fields (like `selectorManager.key`) and `CachedSelectorSlice` is not properly encoded as a result.